### ISSUE TITLE
Click behavior: dashboard and saved question links are not preserved during export / import

### ISSUE THREAD

**Issue by zbodi74** [metabase/metabase#33699]
### Describe the bug

Click behavior of cards on dashboards is not restored correctly when the link points to a dashboard, or to a saved question.

### To Reproduce

1. In 'Our analytics': create a dashboard, name it as `Dashboard 1`.  Create a saved question (e.g. calculate the number of rows for Sample Data / Accounts), and add it to this dashboard.
2. Edit Dashboard 1, and set click behavior for the card: custom destination, dashboard, pick `Dashboard 1` (the same dashboard) as a destination.
4. Save the dashboard, and verify that clicking the card leads to the same dashboard.
5. Duplicate the dashboard a few times, naming the new dashboards as `Dashboard 2`, ... , `Dashboard 9`.
6. Notice that clicking the card on the duplicated dashboards will all lead to `Dashboard 1`, as expected.
7. Export and import the dashboards into an empty Metabase instance.
8. After the import, when opening any of `Dashboard 1`, ... , `Dashboard 9` and clicking on the card, the link will likely lead to a different card, not `Dashboard 1`.

### Expected behavior

The custom destination should not change after the import.

### Logs

n/a

### Information about your Metabase installation

```JSON
- Metabase version: v1.47.0
```

### Severity

Reported by a customer

### Additional context

I believe this happens because the export yaml is saved with the numeric id of the destination which is not guaranteed to stay the same after an import. We should use the entity id instead.
```
  visualization_settings:
    click_behavior:
      type: link
      linkType: dashboard
      targetId: 1
```

The exported data also has a similar structure when the destination is a saved question. I have not tested this specifically, but the same issue is likely happening with this link type as well:
```
    click_behavior:
      type: link
      linkType: question
      targetId: 3
```

**Comment by luizarakaki** [comment]
Reopened because of conflicts

**Comment by bshepherdson** [comment]
#35717 is the original fix for master; backported by #35864.

We discovered a new issue #35880 which @piranha fixed in #35911. When that is backported too, this will be done.

**Comment by piranha** [comment]
Hopefully all fixed with #35913
